### ISSUE TITLE
Add support for RUNS metric

### ISSUE THREAD

**Issue by jim-smith** [AI-SDC/SACRO-ML#220]
Basic idea:
dataset D split into training set Tr and test set Te
1. concatenate train and test sets (so the top |Tr| records come from the training set and the bottom |Te| from the test set
2. pass that dataset into an attack model and get an array C (confidences) of size |D| holding attackers' confidence that each record is in the training set.
3. Add second column *label* to C with C[i][1]  = 1 I<=|Tr| else 0
4. Sort C by first column (confidence)
5. Use the runs test to see whether the number of runs (sequences of the same value) within labels column of C is different to what you would expect if they were randomly distributed.

[statsmodels](https://www.statsmodels.org/stable/generated/statsmodels.sandbox.stats.runs.runstest_1samp.html) has an implementation, but we might want to adapt it to the two-tail version: [wikepedia](https://en.wikipedia.org/wiki/Wald–Wolfowitz_runs_test) 

There's lots of good advice in [here](https://www.ncss.com/wp-content/themes/ncss/pdf/Procedures/NCSS/Analysis_of_Runs.pdf) and simon has pointed out that we need to decide what to do in the case of ties when we sort by probability
